WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -979,30 +987,47 @@ private void updateLocationText(String locationName) {
     }
 
     private void showPostExcerptDialog() {
-        PostExcerptDialogFragment dialog = PostExcerptDialogFragment.newInstance(getPostExcerptFromTextView());
-        dialog.setPostExcerptDialogListener(new PostExcerptDialogFragment.PostExcerptDialogListener() {
-            @Override
-            public void onPostExcerptUpdated(String postExcerpt) {
-                if (TextUtils.isEmpty(postExcerpt)) {
-                    setPostExcerptToNotSet();
-                } else {
-                    mExcerptTextView.setText(postExcerpt);
-                }
-            }
-        });
+        String currentExcerpt = getTextFromTextView(mExcerptTextView);
+        PostSettingsInputDialogFragment dialog = PostSettingsInputDialogFragment.newInstance(
+                currentExcerpt, getString(R.string.post_excerpt), getString(R.string.post_excerpt_dialog_hint));
+        dialog.setPostSettingsInputDialogListener(
+                new PostSettingsInputDialogFragment.PostSettingsInputDialogListener() {
+                    @Override
+                    public void onInputUpdated(String input) {
+                        setTextIfPresent(mExcerptTextView, input);
+                    }
+                });
+        dialog.show(getFragmentManager(), null);
+    }
+
+    private void showSlugDialog() {
+        String currentSlug = getTextFromTextView(mSlugTextView);
+        PostSettingsInputDialogFragment dialog = PostSettingsInputDialogFragment.newInstance(
+                currentSlug, getString(R.string.post_slug), getString(R.string.post_slug_dialog_hint));
+        dialog.setPostSettingsInputDialogListener(
+                new PostSettingsInputDialogFragment.PostSettingsInputDialogListener() {
+                    @Override
+                    public void onInputUpdated(String input) {
+                        setTextIfPresent(mSlugTextView, input);
+                    }
+                });
         dialog.show(getFragmentManager(), null);
     }
 
-    private String getPostExcerptFromTextView() {
-        String excerpt = mExcerptTextView.getText().toString();
-        if (excerpt.equals(getString(R.string.not_set))) {
-            excerpt = "";
+    private String getTextFromTextView(TextView textView) {
+        String text = textView.getText().toString();
+        if (text.equals(getString(R.string.not_set))) {
```

Comment:
Fixed in 5bc2572717bb66edbf53f3bb3eec82488b5802be. If the user sets the `Excerpt` or the `Slug` to `Not Set` it'll work, but, it'll look a little bit weird since we use `Not Set` for other stuff. But this is such an edge case that I don't think we should worry about changing the design for this. If you feel otherwise, let's ping a designer!

P.S: If someone uses `Not Set`, I think they deserve the confusion! :trollface: